app/src/main/java/com/dessalines/thumbkey/ui/screens/AbbreviationsScreen.kt

```diff
@@ -0,0 +1,226 @@
+                    abbreviationToEdit = null
+                },
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun AbbreviationItem(
+    abbreviation: Abbreviation,
+    onEdit: (Abbreviation) -> Unit,
+    onDelete: (Abbreviation) -> Unit,
+) {
+    Card(
+        modifier =
+            Modifier
+                .fillMaxWidth()
+                .padding(vertical = 4.dp),
+        onClick = { onEdit(abbreviation) },
+    ) {
+        Row(
+            modifier =
+                Modifier
+                    .padding(16.dp)
+                    .fillMaxWidth(),
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            Column {
+                Text(
+                    text = abbreviation.abbreviation,
+                    style = MaterialTheme.typography.titleMedium,
+                )
+                Text(
+                    text = abbreviation.expansion,
+                    style = MaterialTheme.typography.bodyMedium,
+                )
+            }
+            IconButton(onClick = { onDelete(abbreviation) }) {
+                Icon(Icons.Default.Delete, contentDescription = "Delete")
+            }
+        }
+    }
+}
+
+@Composable
+fun AddEditAbbreviationDialog(
+    abbreviation: Abbreviation?,
+    onDismiss: () -> Unit,
+    onSave: (String, String) -> Unit,
+) {
+    var abbr by remember { mutableStateOf(abbreviation?.abbreviation ?: "") }
+    var expansion by remember { mutableStateOf(abbreviation?.expansion ?: "") }
+
+    AlertDialog(
+        onDismissRequest = onDismiss,
+        title = {
+            Text(
+                text =
+                    if (abbreviation == null) {
+                        stringResource(R.string.add_abbreviation)
+                    } else {
+                        stringResource(R.string.edit_abbreviation)
+                    },
+            )
+        },
+        text = {
+            Column {
+                OutlinedTextField(
+                    value = abbr,
+                    onValueChange = { abbr = it },
+                    label = { Text(stringResource(R.string.abbreviation)) },
+                    singleLine = true,
+                    modifier = Modifier.fillMaxWidth(),
+                )
+                Spacer(modifier = Modifier.height(8.dp))
+                OutlinedTextField(
+                    value = expansion,
+                    onValueChange = { expansion = it },
+                    label = { Text(stringResource(R.string.expansion)) },
+                    singleLine = true,
+                    modifier = Modifier.fillMaxWidth(),
+                )
+            }
+        },
+        confirmButton = {
+            TextButton(
+                onClick = {
+                    if (abbr.isNotBlank() && expansion.isNotBlank()) {
+                        onSave(abbr, expansion)
+                    }
```

Comment:
Instead of this being inside the onClick, have this check use the `enabed` on the text button.  `enabled = abbr.isNotBlank() && ...`

That in addition to the `isError's` on the 2 outlinedtextfield's above, should be good enough.